hbase-http/src/main/java/org/apache/hadoop/hbase/http/HttpServer.java

```diff
@@ -154,6 +153,20 @@ public class HttpServer implements FilterContainer {
   public static final String NO_CACHE_FILTER = "NoCacheFilter";
   public static final String APP_DIR = "webapps";
 
+  public static final String METRIC_SERVLETS_CONF_KEY = "hbase.http.metrics.servlets";
+  public static final String METRICS_SERVLETS_DEFAULT[] = { "jmx", "metrics", "prometheus" };
+  private static final Map<String, ServletConfig> METRIC_SERVLETS =
+    new HashMap<String, ServletConfig>() {
```

Comment:
Use ImmutableMap.Builder in guava. In hbase we have a shaded guava, under the org.apache.hbase.thirdparty.com.google package. The current code will lead to a error prone warning.